llmfoundry/data/packing.py

```diff
@@ -58,7 +57,7 @@ def efficiency(self) -> float:
 
     def __call__(
             self,
-            examples: List[Dict[str, torch.Tensor]]) -> Dict[str, torch.Tensor]:
+            examples: Dict[str, torch.Tensor]) -> Dict[str, torch.Tensor]:
```

Comment:
ah yes that's right! thanks for catching that

Reply:
hmmm I think the type isn't necessarily consistent because during profiling the type of examples is `Dict[str, torch.Tensor]` while self.base_collator is `lambda x: x`

Reply:
maybe we can refactor this out into two functions. I think that'll be cleaner also:
```
def __call__(self, examples: List[Dict[str, torch.Tensor]]):
    batch = self.base_collator(examples)
    return self.pack(batch)

def pack(self, examples: Dict[str, torch.Tensor]):
    # assertions, packing logic
```
and then in profiling we can just called BinPackCollator.pack

Reply:
Updated to do this refactor. (It was needed to appease the pyright gods anyway)

Reply:
Sorry for not chiming in sooner. I'm on board with any refactor that makes sense to you and whatever aligns this code with the auto packing script such that pyright is appeased.